sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsException.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+import com.azure.core.exception.AzureException;
+
+import java.util.Locale;
+
+/**
+ * General exception for Text Analytics related failures.
+ */
+public class TextAnalyticsException extends AzureException {
+    private static final long serialVersionUID = 21436310107606058L;
+
+    private final String errorCodeValue;
+    private final String target;
+
+    /**
+     * Initializes a new instance of the TextAnalyticsException class.
+     * @param message Text containing any additional details of the exception.
+     * @param errorCodeValue The service returned error code value.
+     * @param target The target for this exception.
+     */
+    public TextAnalyticsException(String message, String errorCodeValue, String target) {
+        super(message);
+        this.errorCodeValue = errorCodeValue;
+        this.target = target;
+    }
+
+    @Override
+    public String getMessage() {
+        String baseMessage = super.getMessage();
+
+        if (this.errorCodeValue == null) {
+            return super.getMessage();
```

Comment:
`baseMessage` could be returned here instead of retrieving it from the super class again.